qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -0,0 +1,526 @@
+        # output dimensions from `_get_compose_dims` as the dimensions of the
+        # Clifford object can't be changed by composition
+        self._get_compose_dims(other, qargs, front)
+
+        # Pad other with identities if composeing on subsystem
+        other = self._pad_with_identity(other, qargs)
+        return self._compose_clifford(other, front=front)
+
+    def dot(self, other, qargs=None):
+        """Return the right multiplied operator self * other.
+
+        Args:
+            other (Clifford): an operator object.
+            qargs (list or None): a list of subsystem positions to apply
+                                  other on. If None apply on all
+                                  subsystems [default: None].
+
+        Returns:
+            Clifford: The operator self * other.
+
+        Raises:
+            QiskitError: if operators have incompatible dimensions for
+                         composition.
+        """
+        return super().dot(other, qargs=qargs)
+
+    def tensor(self, other):
+        """Return the tensor product operator self ⊗ other.
+
+        Args:
+            other (Clifford): a operator subclass object.
+
+        Returns:
+            Clifford: the tensor product operator self ⊗ other.
+        """
+        return self._tensor_product(other, reverse=False)
+
+    def expand(self, other):
+        """Return the tensor product operator other ⊗ self.
+
+        Args:
+            other (Clifford): an operator object.
+
+        Returns:
+            Clifford: the tensor product operator other ⊗ self.
+        """
+        return self._tensor_product(other, reverse=True)
+
+    # ---------------------------------------------------------------------
+    # Representation conversions
+    # ---------------------------------------------------------------------
+
+    def to_dict(self):
+        """Return dictionary represenation of Clifford object."""
+        return {
+            "stabilizer": self.stabilizer.to_labels(),
+            "destabilizer": self.destabilizer.to_labels()
+        }
+
+    @staticmethod
+    def from_dict(obj):
+        """Load a Clifford from a dictionary"""
+        destabilizer = StabilizerTable.from_labels(obj.get('destabilizer'))
+        stabilizer = StabilizerTable.from_labels(obj.get('stabilizer'))
+        return Clifford(destabilizer + stabilizer)
+
+    def to_matrix(self):
+        """Convert operator to Numpy matrix."""
+        return self.to_operator().data
+
+    def to_operator(self):
+        """Convert to an Operator object."""
+        return Operator(self.to_instruction())
+
+    def to_circuit(self):
+        """Return a QuantumCircuit implementing the Clifford."""
+        return decompose_clifford(self)
+
+    def to_instruction(self):
+        """Return a Gate instruction implementing the Clifford."""
+        return self.to_circuit().to_gate()
+
+    @staticmethod
+    def from_instruction(instruction):
```

Comment:
I'd call this `from_circuit` to be consistent with (future!) plans to unify things under the banner of circuits.

Reply:
also there are probably other instances of this function in other parts of quantum_info. they should all just talk about converting to/from that object and circuits. One side is simulation and the other is synthesis.

Reply:
Most of the operators / states have a from instruction that I could change to from_circuit. Will have to add deprecation warnings to from_instruction for `Statevector` and `DensityMatrix`